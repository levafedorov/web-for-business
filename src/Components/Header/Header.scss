
@mixin flex($just:center, $al:center){
    display: flex;
      justify-content: $just; 
      align-items: $al;
   }
 
   @mixin flex-self($just:center, $al:center){
      justify-self: $just; 
      align-self: $al;
   }

   @mixin myriad($size:1em, $height:1em, $weight:normal){
    font-family: 'Myriad Pro Regular', 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
    font-size: $size;
    line-height: $height;
    font-weight: $weight;
}

@mixin baskerville($size:1em, $height:1em, $weight:normal){
   font-family: 'Baskerville Regular', 'Times New Roman', Times, serif;
   font-size: $size;
   line-height: $height;
   font-weight: $weight;
}

@mixin athelas($size:1em, $height:1em, $weight:normal){
   font-family: 'Athelas Regular', Georgia, 'Times New Roman', Times, serif;
   font-size: $size;
   line-height: $height;
   font-weight: $weight;
}

@mixin quatro($size:1em, $height:1em, $weight:normal){
   font-family: 'Quatro Regular' 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
   font-size: $size;
   line-height: $height;
   font-weight: $weight;
}






.main-header{box-sizing: border-box;
              width: 100%;
             padding: 10px;
             position: relative;
             z-index: 1001;
            }

 .main-header .positioner{
     height: 120px;
 }  

.logo-and-themes{
    @include flex();
}

.logo-and-themes .positioner{
    @include flex(space-between, center);
}

.theme-bar{
    width: 100px;
    @include flex(space-between, center);
}

.main-header{
     .logo{
         @include myriad(1.3em, 1.3em);
     }
}



@media screen and (max-width: 500px) and (max-height: 700px){
    .main-header .positioner {
        height: 60px;
    }
}