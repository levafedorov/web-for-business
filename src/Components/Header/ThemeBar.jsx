import React, {Component} from 'react';
import ThemeContext from '../PageContexts/PageThemes';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Button from '../Buttons/Button';
import PropTypes from 'prop-types';


export default class ThemeBar extends Component{
   
 static contextType = ThemeContext;
 
 btnHanndler = (ev) => {
      const theme = ev.currentTarget.dataset.theme;
      this.context.change(theme);
 }


    render(){
       const {btnHanndler} = this;
        return <div className='theme-bar'>
            <Button theme='default' fn={btnHanndler} class={'rounded small '}><FontAwesomeIcon icon="circle" color="grey" size="xs"/></Button>
            <Button theme='dark'   fn={btnHanndler} class={'rounded small '}><FontAwesomeIcon icon="moon" color="grey"/></Button>
            <Button theme='light' fn={btnHanndler} class={'rounded small '}><FontAwesomeIcon icon="sun" color="grey"/></Button>
        </div>
    }
}


