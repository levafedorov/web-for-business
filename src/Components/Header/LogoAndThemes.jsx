import React, {Component} from 'react';
import ThemeBar from './ThemeBar';
import Positioner from '../Page/Positioner';

export default class LogoAndThemes extends Component{
   
    render(){
        return <section className="logo-and-themes">
                  <Positioner>
                        <div className="logo">YOURCOMPANYNAME</div>
                        <ThemeBar/>
                  </Positioner>
             </section>
    }
}