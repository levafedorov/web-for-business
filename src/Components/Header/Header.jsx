import React, {Component} from 'react';
import LogoAndThemes from './LogoAndThemes';
import './Header.scss';



export default class Header extends Component{

   






    render(){
    
    return <header className='main-header'>
               <LogoAndThemes />
          </header>     
    }
}