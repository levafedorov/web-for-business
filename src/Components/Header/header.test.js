import React from 'react';
import {shallow} from 'enzyme';
import Header from './Header';



test('Should header render', () => {
        const component = shallow(<Header />);
        expect(component).toMatchSnapshot();
    });

 
