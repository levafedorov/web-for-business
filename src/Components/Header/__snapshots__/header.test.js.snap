// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should header render 1`] = `
<header
  className="main-header"
>
  <LogoAndThemes />
</header>
`;
